src/plugins/links/public/components/dashboard_link/dashboard_link_component.tsx

```diff
@@ -78,29 +50,22 @@ export const DashboardLinkComponent = ({
    */
   const [dashboardTitle, dashboardDescription] = useMemo(() => {
     return link.destination === parentDashboardId
-      ? [parentDashboardInput?.title, parentDashboardInput?.description]
-      : [destinationDashboard?.attributes.title, destinationDashboard?.attributes.description];
-  }, [link.destination, parentDashboardId, parentDashboardInput, destinationDashboard]);
-
-  /**
-   * Memoized link information
-   */
-  const linkLabel = useMemo(() => {
-    return link.label || (dashboardTitle ?? DashboardLinkStrings.getDashboardErrorLabel());
-  }, [link, dashboardTitle]);
+      ? [dashboardContainer.getTitle(), dashboardContainer.getDescription()]
+      : [link.label ?? link.title, link.description];
+  }, [link, parentDashboardId, dashboardContainer]);
```

Comment:
If we use the title publishing subject here (from the parent) instead, I think this would resolve https://github.com/elastic/kibana/pull/178670#discussion_r1583499739? Maybe? :eyes: